src/serac/numerics/functional/tensor.hpp

```diff
@@ -1170,13 +1172,26 @@ SERAC_HOST_DEVICE bool is_symmetric_and_positive_definite(tensor<double, 3, 3> A
 }
 
 /**
- * @brief Solves Ax = b for x using Gaussian elimination with partial pivoting
- * @param[in] A The coefficient matrix A
- * @param[in] b The righthand side vector b
- * @note @a A and @a b are by-value as they are mutated as part of the elimination
+ * @brief Representation of an LU factorization
+ *
+ * The entries of P mean row i of the matrix was exchanged with row P[i].
  */
 template <typename T, int n>
-SERAC_HOST_DEVICE constexpr tensor<T, n> linear_solve(tensor<T, n, n> A, const tensor<T, n> b)
+struct LuFactorization {
+  tensor<int, n>  P;  ///< Row permutation indices due to partial pivoting
+  tensor<T, n, n> L;  ///< Lower triangular factor
+  tensor<T, n, n> U;  ///< Upper triangular factor
+};
```

Comment:
does `L` or `U` have 1s on the diagonal? If so, that would be good to note here

Reply:
Agreed. I added this comment. I also noted it in the Doxygen block for the function definition of `lu()`.